packages/consumer/src/mailClient.ts

```diff
@@ -0,0 +1,34 @@
+import { createTransport, Transporter } from "nodemailer";
+import { Email } from "@email-service/common";
+import { config } from "./config/config";
+
+export class MailClient {
+  private transporter: Transporter | undefined;
+  setupTransporter(accessToken: string, refreshToken: string) {
+    this.transporter = createTransport({
+      service: "gmail",
+      auth: {
+        type: "OAuth2",
+        accessToken,
+        user: config.FROM_EMAIL,
+        clientId: config.OAUTH_CLIENT_ID,
+        clientSecret: config.OAUTH_CLIENT_SECRET,
+        refreshToken: refreshToken,
+      },
+    });
+    console.log("Transporter created!");
+  }
+
+  public sendEmail() {
+    if (!this.transporter) {
+      throw new Error("Transporter is not initialized");
+    }
+    return async (email: Email) => {
+      await this.transporter!.sendMail({
+        ...email,
+        from: `${email.from || "Kir-Dev"} <${config.FROM_EMAIL}>`,
```

Comment:
Is this needed explicitly? Doesn't the email client automatically constructs this syntax from the email and sender?

Reply:
it's important that we can specify the sender. For example, emails sent by Konzisite will show up as if Konzisite sent them and not Kir-Dev

Reply:
We can, that's just a fallback